dotcom-rendering/src/web/components/ShowMore.importable.tsx

```diff
@@ -0,0 +1,125 @@
+import { css } from '@emotion/react';
+import {
+	error as errorColor,
+	fontWeights,
+	from,
+	space,
+} from '@guardian/source-foundations';
+import { Button, SvgMinus, SvgPlus } from '@guardian/source-react-components';
+import { useState } from 'react';
+import { useApi } from '../lib/useApi';
+
+function decideButtonText({
+	showMore,
+	loading,
+	displayName,
+}: {
+	showMore: boolean;
+	loading: boolean;
+	displayName: string;
+}) {
+	if (showMore && loading) return 'Loading';
+	if (showMore) return 'Less';
+	return `More ${displayName}`;
+}
+
+function insertHtml(html: string, collectionId: string) {
+	try {
+		const placeholder = document.querySelector<HTMLElement>(
+			`[data-show-more-placeholder="${collectionId}"]`,
+		);
+		/**
+		 * it is expected that the value of `html` will be a server-rendered
+		 * instance of the ExtraCardsContainer component
+		 */
+		// @ts-expect-error -- We want to catch this error if the element is missing
+		placeholder.innerHTML = html;
+	} catch (e) {
+		// todo: `window` is throwing a referenceError; find a fix
```

Comment:
@OllysCoding I mentioned this in chat but just realised I didn't flag it here. I feel like we could leave this for a separate issue, but if you think otherwise (or if you know how to fix this?) then lmk?